### PR TITLE
CEDS 1411: Added Exports CacheModel to GoodsShipment Mapping. This did not include GovernmentAgencyGoodsItem

### PR DESCRIPTION
**Pull request by mattclark-zerogravit**

### CODE REVIEW

File: app/services/cache/mapping/declaration/DeclarationBuilder.scala [hunk 1]
```diff
@@ -30,21 +29,15 @@ object DeclarationBuilder {
...
 
     declaration.setFunctionCode(FunctionCodeBuilder.build(defaultFunctionCode))
-    declaration.setFunctionalReferenceID(createDeclarationFunctionalReferenceIDType(exportsCacheModel))
 
-    val typeCode: Option[DeclarationTypeCodeType] = exportsCacheModel.additionalDeclarationType.map(
-      additionalDeclarationType =>
-        TypeCodeBuilder.createTypeCode(additionalDeclarationType, exportsCacheModel.dispatchLocation)
-    )
+    FunctionalReferenceIdBuilder.buildThenAdd(exportsCacheModel, declaration)
+    TypeCodeBuilder.buildThenAdd(exportsCacheModel, declaration)
```
Comment (PauloPortugal) [inline app/services/cache/mapping/declaration/DeclarationBuilder.scala]: :+1:

File: app/services/mapping/declaration/AgentBuilder.scala [hunk 2]
```diff
@@ -32,7 +33,14 @@ object AgentBuilder {
       .map(data => createAgent(data))
       .orNull
 
-  def createAgent(data: RepresentativeDetails): Declaration.Agent = {
+  def buildThenAdd(exportsCacheModel: ExportsCacheModel, declaration: Declaration): Unit =
+    exportsCacheModel.parties.representativeDetails.foreach { representativeDetails =>
```
Comment (PauloPortugal) [inline app/services/mapping/declaration/AgentBuilder.scala]: :+1:

File: app/services/mapping/declaration/FunctionalReferenceIdBuilder.scala [hunk 3]
```diff
@@ -33,6 +35,16 @@ object FunctionalReferenceIdBuilder {
     referenceId
   }
 
+  def buildThenAdd(exportsCacheModel: ExportsCacheModel, declaration: Declaration) {
+    exportsCacheModel.consignmentReferences.foreach(references => {
+      val declarationFunctionalReferenceIdType = new DeclarationFunctionalReferenceIDType()
```
Comment (PauloPortugal) [inline app/services/mapping/declaration/FunctionalReferenceIdBuilder.scala]: I would move the instantiation inside the if statement. 
We only want to create an instance if we have an `LRN`

File: app/services/mapping/goodsshipment/DestinationBuilder.scala [hunk 4]
```diff
@@ -32,6 +33,13 @@ object DestinationBuilder {
       .map(data => createExportCountry(data.countryOfDestination))
       .orNull
 
+  def buildThenAdd(exportsCacheModel: ExportsCacheModel, goodsShipment: GoodsShipment) =
+    exportsCacheModel.locations.destinationCountries.foreach { destinationCountries =>
+      {
```
Comment (PauloPortugal) [inline app/services/mapping/goodsshipment/DestinationBuilder.scala]: I think we do not need these curly braces here

File: app/services/mapping/goodsshipment/GoodsShipmentBuilder.scala [hunk 5]
```diff
@@ -40,4 +41,21 @@ object GoodsShipmentBuilder {
 
     goodsShipment
   }
+
+  def build(exportsCacheModel: ExportsCacheModel): GoodsShipment = {
+    val goodsShipment = new GoodsShipment()
+
+    GoodsShipmentNatureOfTransactionBuilder.buildThenAdd(exportsCacheModel.natureOfTransaction, goodsShipment)
```
Comment (PauloPortugal) [inline app/services/mapping/goodsshipment/GoodsShipmentBuilder.scala]: :+1:

File: app/services/mapping/goodsshipment/DestinationBuilder.scala (hunk 4 above)
Comment (mattclark-zerogravit) [reply app/services/mapping/goodsshipment/DestinationBuilder.scala]: changed

File: app/services/mapping/declaration/FunctionalReferenceIdBuilder.scala (hunk 3 above)
Comment (mattclark-zerogravit) [reply app/services/mapping/declaration/FunctionalReferenceIdBuilder.scala]: yep agreed, have changed.

**Pull request merged**